include/mxnet/ndarray.h

```diff
@@ -513,6 +515,7 @@ class NDArray {
     // We can't reuse memory in a view.
     CHECK(!IsView());
     NDArray ret = *this;
+    ret.byte_offset_ = byte_offset_ + byte_offset;
```

Comment:
this might be problematic. The way of testing if an array is a view uses both `reuse_` and `byte_offset_`. https://github.com/apache/incubator-mxnet/blob/master/include/mxnet/ndarray.h#L147
if `reuse_` is true, the array can't be a view. I wonder if setting `byte_offset_`  to non-zero can cause problems.